packages/scandipwa/src/component/CategoryConfigurableAttributes/CategoryConfigurableAttributes.component.js

```diff
@@ -9,15 +9,36 @@
  * @link https://github.com/scandipwa/base-theme
  */
 
+import PropTypes from 'prop-types';
+
 import ExpandableContent from 'Component/ExpandableContent';
 import ExpandableContentShowMore from 'Component/ExpandableContentShowMore';
 import ProductAttributeValue from 'Component/ProductAttributeValue/ProductAttributeValue.component';
 // eslint-disable-next-line max-len
 import ProductConfigurableAttributes from 'Component/ProductConfigurableAttributes/ProductConfigurableAttributes.component';
+import { CategoryFragment } from 'Type/Category';
 import { formatPrice } from 'Util/Price';
 
 /** @namespace Component/CategoryConfigurableAttributes/Component */
 export class CategoryConfigurableAttributes extends ProductConfigurableAttributes {
+    static propTypes = {
+        childrenCategories: PropTypes.arrayOf(PropTypes.shape(CategoryFragment))
+    };
+
+    renderSubCategories(option) {
+        const optionWithSubcategories = option;
+        const { childrenCategories } = this.props;
+        const { attribute_values } = option;
+        const childrenCategoryIds = childrenCategories.map((category) => category.id.toString());
+        const subCategoriesIds = attribute_values.filter((item) => childrenCategoryIds.includes(item));
+        optionWithSubcategories.attribute_values = subCategoriesIds;
```

Comment:
On the second thought, this logic should be refactored a bit more.
No data processing should happen in the component. Let's move this logic to a separate place, preferably - outside the component (e.g. container or util)

Reply:
I have reviewed not noticing that Zhan has already done it 😄 
Anyways, please provide this little change and then - all good.